### ISSUE TITLE
Why do the steps to sum n vectors seem to form n-dimentional cuboids?

### ISSUE THREAD

**Issue by danicruz0415** [leios/SoME_Topics#132]
Hi!, 

I was really doubtful about posting this, cause I stumbled upon it but I didn't know if its trivially dumb or can be interesting in some sort of way. I was taught of the "parallelogram rule" for adding vectors, which consists in drawing the two vectors that you want to add, then draw the first one starting from the end of the second one and vice versa. This forms a parallelogram as shown in the image (taken from https://mathworld.wolfram.com/VectorAddition.html):
![image](https://user-images.githubusercontent.com/107332190/173213330-452508c0-f124-4413-a73e-4d83440a4776.png)

So far so good, but I wondered, what happens if instead of 2 vectors I do this with 3? What kind of shape would it form? It turns out it forms a sort of stretched cube! 
![image](https://user-images.githubusercontent.com/107332190/173213354-7ff2f7b8-1029-4db9-b6d9-55825134ba02.png)

If we consider a parallelogram as a stretched square, then I thought, would having 4 form the projection of a tesseract? and it does! (although not the one with perspective we sometimes are used to see, but an isometric one like the one shown [here](https://en.wikipedia.org/wiki/Tesseract#Projections))
![image](https://user-images.githubusercontent.com/107332190/173213556-9e9358d1-5402-4d2c-9b38-7af45fbe6008.png)

For me its not completely obvious why it should be that this specific shape would show up. I mean for me makes sense that adding 3 3-dimentional vectors would form a stretched cube, but we are taking only 2 dimensional vectors. I know this probably has to do with the way projections are calculated, but I don't know if this has any ground to become an interesting "story to tell", or an interesting way to introduce a topic.

I know this can be trivial or even dumb, but I've seen so much interesting topics emerging from almost "stupid looking" questions that I did not want to risk missing out if this had 1% chance of becoming one

## About the author

I am a software developer with interests in math

### More details

I leave an interactive version of this if someone wants to play around with it (click on the plus button to add a vector, and you can drag the vectors (from the head) with your mouse to see how the shape changes (you can only drag the "bright" version of the vector))
https://editor.p5js.org/danicruz0415/sketches/slD7Ux6UL

### Contact details

Respond here

**Comment by GaussianZero** [comment]
To me this seems related to the concept of "basis" for linear transformations (i.e. you're starting at the origin and each of these vectors is one of your "axes". this concept can tie in to many pieces of linear algebra). Overall, like you mention this is kind of a simple/trivial idea, but IMO it can serve as a foundation for many interesting topics.

Good luck with the idea!

**Comment by Arongil** [comment]
Hey, like you said this is a simple idea, but I believe it can lead into very interesting mathematics. One place to look might also be n-dimensional hypercubes. The vertices of such an object can be described as binary strings of length n.

Ex: for a square, the four corners are 00, 01, 10, 11.
Ex: for a cube, the eight corners are 000, 001, 010, 011, 100, 101, 110, 111

There are a lot of interesting graph theory and combinatorics questions you can ask from there. For example, how many walks are there from one corner to an opposite corner? But even your original question, which is this: "Why do binary strings form hypercubes?" That could work as a video title! And to push it a bit further, what shapes do strings in base 3 make? I think hypertriangles, but I haven't thought too much and this could be interesting to explore. What about strings in base m? How many vertices, sides, and faces do these objects have? How do they connect to other areas of math?

No matter your starting point, if you dig a little you get into interesting math. I'd encourage you to make a video for SoME#2. Don't worry whether your initial idea to is too "basic" -- it's all about the how and why, not so much the what. Good luck!

**Comment by en-GB** [comment]
theyre called zonohedra. i think de brujin used them to study the penrose tiling
but theyre especially useful for computer graphics/computational geometry.

i use them as primitive shapes in collision detection and for all kinds of shader tricks
like efficiently rendering sand and other glinty materials.

theyre used alot for mathematical art. (https://www.youtube.com/watch?v=_PSdVX02Vbs)

there is also a neat connection with 3b1b's counting video (https://www.youtube.com/watch?v=bOXCLR3Wric).
the generating function he uses is the fourier transform of a particular one dimensional zonohedron.
or atleast if you plug in the right x.

**Comment by danicruz0415** [comment]
> To me this seems related to the concept of "basis" for linear transformations (i.e. you're starting at the origin and each of these vectors is one of your "axes". this concept can tie in to many pieces of linear algebra). Overall, like you mention this is kind of a simple/trivial idea, but IMO it can serve as a foundation for many interesting topics.
> 
> Good luck with the idea!

Yes @GaussianZero I think I can relate this to transformations, after all, to program the interactive version what I did was to draw each vector being transformed by all the others in all possible orders. I'll see how can I make this relation more clear and see if I can redact something interesting

**Comment by danicruz0415** [comment]
> Hey, like you said this is a simple idea, but I believe it can lead into very interesting mathematics. One place to look might also be n-dimensional hypercubes. The vertices of such an object can be described as binary strings of length n.
> 
> Ex: for a square, the four corners are 00, 01, 10, 11. Ex: for a cube, the eight corners are 000, 001, 010, 011, 100, 101, 110, 111
> 
> There are a lot of interesting graph theory and combinatorics questions you can ask from there. For example, how many walks are there from one corner to an opposite corner? But even your original question, which is this: "Why do binary strings form hypercubes?" That could work as a video title! And to push it a bit further, what shapes do strings in base 3 make? I think hypertriangles, but I haven't thought too much and this could be interesting to explore. What about strings in base m? How many vertices, sides, and faces do these objects have? How do they connect to other areas of math?
> 
> No matter your starting point, if you dig a little you get into interesting math. I'd encourage you to make a video for SoME#2. Don't worry whether your initial idea to is too "basic" -- it's all about the how and why, not so much the what. Good luck!

That's really interesting @Arongil, I'll check the topic to see how complex would be to explaining using what I know about vectors. Thanks!

**Comment by danicruz0415** [comment]
> theyre called zonohedra. you can use them for lots of mathy things like constructing non-periodic tilings but theyre especially useful for computer graphics and computational geometry.
> 
> i use them as primitive shapes in collision detection and for all kinds of shader tricks like efficiently rendering sand and other glinty materials.
> 
> theyre used alot for mathematical art. (https://www.youtube.com/watch?v=_PSdVX02Vbs)
> 
> there is also a neat connection with 3b1b's counting video (https://www.youtube.com/watch?v=bOXCLR3Wric). the generating function he uses is the fourier transform of a particular one dimensional zonohedron. or atleast if you plug in the right x.

Hey! I haven't heard of those before. They look like a really cool combination between math and something that could become art. Ill check them out 😃
